### ISSUE TITLE
Add validateOnlyAssignment option to RequireThis check

### ISSUE THREAD

**Issue by octylFractal** [checkstyle/checkstyle#11176]
**Is your feature request related to a problem? Please describe.**
I would only like to enforce `this.` on assignments, rather than on all field accesses. This is similar in desire to #2362, making the check useful without being overly harsh in cases where it doesn't matter.

**Describe the solution you'd like**
```bash
$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
"-//Puppy Crawl//DTD Check Configuration 1.3//EN"
"http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="error"/>
    <property name="fileExtensions" value="java, properties, xml"/>
    <module name="TreeWalker">
        <module name="RequireThis">
            <property name="validateOnlyOverlapping" value="false"/>
            <property name="validateOnlyAssignment" value="true"/> <!-- default value: false -->
        </module>
    </module>
</module>
$ cat RequireThisAssignOnly.java 
class RequireThisAssignOnly {
    long field;
    public void setField(long fieldValue) {
        this.field = fieldValue; // ok
        field = 0; // violation: Reference to instance variable 'field' needs "this.". [RequireThis]
        System.err.println(field); // ok
        System.err.println(field = 0); // violation: Reference to instance variable 'field' needs "this.". [RequireThis]
    }
}
```

**Comment by nmancus1** [comment]
While new property is being discussed, desired behavior can be achieved with [SuppressionXpathSingleFilter](https://checkstyle.sourceforge.io/config_filters.html#SuppressionXpathSingleFilter) :


```

➜  cat Test.java
public class Test {
    long field;
    public void setField(long fieldValue) {
        this.field = fieldValue; // ok
        field = 0; // violation: Reference to instance variable 'field' needs "this.". [RequireThis]
        System.err.println(field); // ok
        System.err.println(field = 0); // violation: Reference to instance variable 'field' needs "this.". [RequireThis]
    }
}

➜  cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="RequireThis">
            <property name="validateOnlyOverlapping" value="false"/>
        </module>
        <module name="SuppressionXpathSingleFilter">
            <property name="checks" value="RequireThis"/>
            <property name="query" value="//IDENT[not(ancestor::ASSIGN)]"/>
        </module>
    </module>
</module>

➜  java -jar checkstyle-9.2-all.jar -c config.xml Test.java
Starting audit...
[ERROR] Test.java:5:9: Reference to instance variable 'field' needs "this.". [RequireThis]
[ERROR] Test.java:7:28: Reference to instance variable 'field' needs "this.". [RequireThis]
Audit done.
Checkstyle ends with 2 errors.


```

**Comment by strkkk** [comment]
it looks like this is not a very common demand, and it is better to just use suppression provided above instead of complicating code of the check even more.

is it ok for you @octylFractal ?

**Comment by octylFractal** [comment]
Yes, that works for me.

**Comment by romani** [comment]
we can reconsider this decision if more people would like to have option on this, if somebody also see value in this property please leave comment in this issue.
